code/datums/components/acid.dm

```diff
@@ -15,43 +15,45 @@ GLOBAL_DATUM_INIT(acid_overlay, /mutable_appearance, mutable_appearance('icons/e
 	var/acid_volume
 	/// The maximum volume of acid on the parent [/atom].
 	var/max_volume = INFINITY
-	/// Acid overlay appearance we apply
-	var/acid_overlay
-	/// The ambiant sound of acid eating away at the parent [/atom].
-	var/datum/looping_sound/acid/sizzle
 	/// Used exclusively for melting turfs. TODO: Move integrity to the atom level so that this can be dealt with there.
 	var/parent_integrity = 30
 	/// How far the acid melting of turfs has progressed
 	var/stage = 0
+	/// Acid overlay appearance we apply
+	var/acid_overlay
+	/// The ambient sound of acid eating away at the parent [/atom].
+	var/datum/looping_sound/acid/sizzle
+	/// Particle holder for acid particles (sick)
+	var/obj/effect/abstract/particle_holder/particle_effect
 	/// The proc used to handle the parent [/atom] when processing. TODO: Unify damage and resistance flags so that this doesn't need to exist!
 	var/datum/callback/process_effect
 
-/datum/component/acid/Initialize(acid_power = ACID_POWER_MELT_TURF, acid_volume = 50, acid_overlay = GLOB.acid_overlay)
+/datum/component/acid/Initialize(acid_power = ACID_POWER_MELT_TURF, acid_volume = 50, acid_overlay = GLOB.acid_overlay, acid_particles = /particles/acid)
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
-	//not incompatible, but pointless
+
+	// The parent object cannot have acid. Not incompatible, but should not really happen.
 	var/atom/atom_parent = parent
-	if((acid_power) <= 0 || (acid_volume <= 0))
-		stack_trace("Acid component added to an atom ([atom_parent.type]) with insufficient acid power ([acid_power]) or acid volume ([acid_volume]).")
+	if(atom_parent.resistance_flags & UNACIDABLE)
 		qdel(src)
 		return
```

Comment:
is this valid? the qdel in init bit
![image](https://github.com/tgstation/tgstation/assets/58055496/7c7b9d42-756a-4525-96ae-0d8e7f5047be)

this feels like something we should handle in a wrapper proc for the component add

Reply:
yes it is used in *so* many components, trust me on this one
yes i'll add a wraper in a later PR